compiler/src/iree/compiler/Codegen/Common/TransformDialectJitterPass.cpp

```diff
@@ -0,0 +1,131 @@
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Support/FileUtilities.h"
+
+#define DEBUG_TYPE "iree-transform-dialect-jitter"
+#define DBGS() (llvm::dbgs() << "[" DEBUG_TYPE "]: ")
+
+using namespace mlir;
+
+namespace {
+
+/// Pass declaration.
+/// Jitter pass that applies transform dialect ops for codegen.
+/// This needs to be its own pass because the registration mechanism and ops
+/// available are different than for other jitters.
+class TransformDialectJitterPass
+    : public iree_compiler::TransformDialectJitterBase<
+          TransformDialectJitterPass> {
+ public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    // TODO: this is only necessary to make registry subset happy when running
+    // the lowering to LLVM. The lowering should be changed to stop using the
+    // nested pass manager and this will go away.
+
+    // clang-format off
+    registry.insert<mlir::iree_compiler::IREE::LinalgExt::IREELinalgExtDialect,
+                    mlir::iree_compiler::IREE::Flow::FlowDialect,
+                    arith::ArithDialect,
+                    AffineDialect,
+                    bufferization::BufferizationDialect,
+                    func::FuncDialect,
+                    gpu::GPUDialect,
+                    linalg::LinalgDialect,
+                    linalg::transform::LinalgTransformDialect,
+                    LLVM::LLVMDialect,
+                    pdl::PDLDialect,
+                    pdl_interp::PDLInterpDialect,
+                    scf::SCFDialect,
+                    tensor::TensorDialect,
+                    transform::TransformDialect,
+                    vector::VectorDialect
+        // clang-format on
+        >();
+
+    // TODO: these should be registered by the extension instead, but there is
+    // no support for it in core currently.
+    arith::registerBufferizableOpInterfaceExternalModels(registry);
+    linalg::registerBufferizableOpInterfaceExternalModels(registry);
+    scf::registerBufferizableOpInterfaceExternalModels(registry);
+    bufferization::func_ext::registerBufferizableOpInterfaceExternalModels(
+        registry);
+    tensor::registerBufferizableOpInterfaceExternalModels(registry);
+    vector::registerBufferizableOpInterfaceExternalModels(registry);
+
+    registry.addExtensions<
+        mlir::iree_compiler::IREE::LinalgExt::LinalgExtTransformOpsExtension,
+        transform_ext::StructuredTransformOpsExtension>();
+    iree_compiler::registerTransformDialectCommonExtension(registry);
+    iree_compiler::registerTransformDialectFlowExtension(registry);
+    iree_compiler::registerTransformDialectLLVMCPUExtension(registry);
+    iree_compiler::registerTransformDialectLLVMGPUExtension(registry);
+    linalg::registerTransformDialectExtension(registry);
+  }
+
+  TransformDialectJitterPass() = default;
+
+  void runOnOperation() override {
+    assert(false && "TODO: Impl me");
```

Comment:
Maybe remove the commented part and just return without asserting. No reason to have it in this state if you want to just submit the plumbing....